simulation/tests/test_agents.py

```diff
@@ -241,42 +246,106 @@ def test_households_not_suitable_archetype_are_not_heat_pump_suitable(
 
         assert not unsuitable_archetype_household.is_heat_pump_suitable
 
+    @pytest.mark.parametrize("event_trigger", list(EventTrigger))
     def test_heat_pumps_not_in_heating_system_options_if_household_not_heat_pump_suitable(
         self,
+        event_trigger,
     ) -> None:
 
         unsuitable_household = household_factory(potential_epc=Epc.D)
         model = model_factory()
         assert not HEAT_PUMPS.intersection(
-            unsuitable_household.get_heating_system_options(model)
+            unsuitable_household.get_heating_system_options(
+                model, event_trigger=event_trigger
+            )
         )
 
+    @pytest.mark.parametrize("event_trigger", list(EventTrigger))
     def test_heat_pumps_not_in_heating_system_options_if_household_not_heat_pump_aware(
         self,
+        event_trigger,
     ) -> None:
 
         unaware_household = household_factory(is_heat_pump_aware=False)
         model = model_factory()
         assert not HEAT_PUMPS.intersection(
-            unaware_household.get_heating_system_options(model)
+            unaware_household.get_heating_system_options(
+                model, event_trigger=event_trigger
+            )
         )
 
+    @pytest.mark.parametrize("event_trigger", list(EventTrigger))
     def test_gas_boiler_not_in_heating_system_options_if_household_off_gas_grid(
-        self,
+        self, event_trigger
     ) -> None:
 
         off_gas_grid_household = household_factory(off_gas_grid=True)
         model = model_factory()
         assert not {HeatingSystem.BOILER_GAS}.intersection(
-            off_gas_grid_household.get_heating_system_options(model)
+            off_gas_grid_household.get_heating_system_options(
+                model, event_trigger=event_trigger
```

Comment:
Again, not obligatory, but no reason for the keyword argument with `event_trigger` in these tests.